stdlib/public/core/StringStorage.swift

```diff
@@ -237,14 +297,38 @@ final internal class _StringStorage: _AbstractStringStorage {
 
     return nil
   }
+  
+  @objc(UTF8String)
+  @_effects(readonly)
+  final internal func _utf8String() -> UnsafePointer<UInt8>? {
+    if let contents = asGuts.fastUTF8 {
+      return contents
+    }
```

Comment:
Just return `start`?

Reply:
The 'providesFastUTF8' thing had me under the impression that it wasn't guaranteed, at least for SharedStringStorage. I see we use it unconditionally elsewhere though, so going ahead and making that change.